tests/test_client.py

```diff
@@ -240,5 +244,28 @@ def test_client_unicode_error_py3(client, response):
     # handling the error should work even when utf-8 characters (€) are in the response.
     exception = err.value
     expected = 'Order line 1 is invalid. VAT amount is off. ' \
-        'Expected VAT amount to be €3.47 (21.00% over €20.00), got €3.10'
+               'Expected VAT amount to be €3.47 (21.00% over €20.00), got €3.10'
     assert str(exception) == expected
+
+
+@mock.patch('mollie.api.client.requests.request')
+def test_client_request_timeout(request_mock, client):
+    """Mock requests.request in the client to be able to read if the timeout is in the request call args."""
+    response = mock.Mock(status_code=200)
+    response.json.return_value = {}
+    response.headers.get.return_value = 'application/hal+json'
+    request_mock.return_value = response
+
+    client.set_timeout(300)
+    client.payments.list()
+    assert request_mock.call_args[1]['timeout'] == 300
+
+
+@mock.patch('mollie.api.client.requests.request')
+def test_client_request_timed_out(request_mock, client):
+    """Timeout should raise a RequestError."""
+    request_mock.side_effect = requests.exceptions.ReadTimeout(
+        "HTTPSConnectionPool(host='api.mollie.com', port=443): Read timed out. (read timeout=5)")
```

Comment:
Set the read timeout value in the error response to what we use by default: `(read timeout=10)`